src/V3EmitCHeaders.cpp

```diff
@@ -236,6 +236,21 @@ class EmitCHeader final : public EmitCConstInit {
             puts(itemp->dtypep()->cType(itemp->nameProtect(), false, false));
             puts(";\n");
         }
+
+        puts("\nbool operator==(const " + EmitCBase::prefixNameProtect(sdtypep) + "& rhs){\n");
+        puts("return ");
+        for (const AstMemberDType* itemp = sdtypep->membersp(); itemp;
+             itemp = VN_AS(itemp->nextp(), MemberDType)) {
+            puts(itemp->nameProtect() + " == " + "rhs." + itemp->nameProtect());
+            if (itemp->nextp())
+                puts(" &&\n    ");
+            else
+                puts(";\n");
+        }
+        puts("}");
+        puts("\nbool operator!=(const " + EmitCBase::prefixNameProtect(sdtypep) + "& rhs){\n");
```

Comment:
operator!= should not be needed, C++ defaults to using operator==.

Reply:
Are you sure this is not just c++20 feature?

Here is a small test code, it fails for C++17, but works on C++20 (gcc 12.2.1)
```c++
#include <iostream>

class Some{
    public:
        int m0;
        int m1;
        Some(int m0, int m1) : m0(m0), m1(m1){
        }
        bool operator==(const Some& rhs) const{
            return m0 == rhs.m0 && m1 == rhs.m1;
        }
};
int main(){
    Some some1(4, 5);
    Some some2(1, 2);
    std::cout << (some1 != some2) << "\n";
    return 0;
}
```

From cppreference:
https://en.cppreference.com/w/cpp/language/default_comparisons
https://coliru.stacked-crooked.com/view?id=2ee293e657d89ff4

Reply:
Yes, forgot that is new, you are right we need pre C++20 support.